docs/manual/kinds/ixia-c-one.md

```diff
@@ -0,0 +1,113 @@
+
+The corresponding node in containerlab is identified with `keysight_ixia-c-one` kind in the [topology file](../topo-def-file.md). Upon boot up, it comes up with:
+- management interface `eth0` configured with IPv4/6 addresses as assigned by docker
+- hostname assigned to the node name
+- HTTPS service enabled on port 443 (for client SDK to push configuration and fetch metrics)
+
+## Managing ixia-c-one nodes
+
+ixia-c-one is a `docker:dind` container hosting two kinds of [ixia-c][ixia-c] containers internally:
+- A set of containers acting as API endpoint and managing configuration across multiple test ports
+- A set of containers bound to network interface (created by containerlab), treating it as a test port (i.e. for generating or processing traffic, emulating protocols, etc.)
+
+Request and response to the API endpoint is driven by [Open Traffic Generator API][otg], and can be exercised in following two ways:
+- Using client SDK
+    ```bash
+    # install go from https://go.dev/doc/install since we'll need it to run the test
+    # and setup test environment
+    mkdir tests && cd tests
+    go mod init tests
+    # gosnappi version needs to be compatible to a given release of ixia-c-one and
+    # can be checked from https://github.com/open-traffic-generator/ixia-c/releases
+    go get github.com/open-traffic-generator/snappi/gosnappi@v0.7.18
+    # download a basic IPv4 forwarding test
+    curl -kLO https://raw.githubusercontent.com/open-traffic-generator/snappi-tests/main/scripts/ipv4_forwarding.go
+    # run the test with MAC address obtained in previous step
+    go run ipv4_forwarding.go -dstMac="<MAC address>"
+    ```
+- Using curl and JSON payloads
+    ```bash
+    # fetch configuration that was last pushed to ixia-c-one
+    curl -kL https://clab-ixia-c-ixia-c-one/config
+    # fetch flow metrics
+    curl -kL https://clab-ixia-c-ixia-c-one/results/metrics -d '{"choice": "flow"}'
+    ```
+
+## Interfaces mapping
+ixia-c-one container uses the following mapping for its linux interfaces:
+
+* `eth0` - management interface connected to the containerlab management network
+* The other interfaces are the data interfaces which are created using same name as provided in the containerlab topology yaml file. 
+
+When containerlab launches ixia-c-one node, it will set IPv4/6 addresses as assigned by docker to the `eth0` interface and ixia-c-one node will boot with that addresses configured.
+
+Data interfaces `eth1+` need to be configured with IP addressing manually if needed (as in the Layer3 forwarding test example).
+This is needed when the test port needs to reply to ARP/ND queries from the Device Under Test.
+
+- To configure an IPv4 address on any data link e.g. eth2 (Could be eth1 or eth3 as well, but never on eth0):
+    ```bash
+    docker exec -it clab-ixia-c-ixia-c-one bash -c "./ifcfg add eth2 2.2.2.2 24"
+    ```
```

Comment:
Hi @rudranil-das @ashutshkumr 
I have a question regarding interfaces names in ixia-c-one.

Using the following topology 
```yaml
name: ixia

topology:
  nodes:
    ixia:
      kind: keysight_ixia-c-one
      image: ghcr.io/open-traffic-generator/ixia-c-one:0.0.1-2610
    linux:
      kind: linux
      image: alpine:3

  links:
    - endpoints: ["ixia:eth1", "linux:eth1"]
    - endpoints: ["ixia:eth2", "linux:eth2"]
```

I was expectin to see `eth1/2` interfaces on ixia-c side, but instead I have some veth endpoints:

```
❯ docker exec clab-ixia-ixia ip l    
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: docker0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default 
    link/ether 02:42:e4:32:74:3d brd ff:ff:ff:ff:ff:ff
4: veth4d2f1df@if3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP mode DEFAULT group default 
    link/ether da:79:8d:2e:9c:1b brd ff:ff:ff:ff:ff:ff link-netnsid 2
6: vethddc45d5@if5: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP mode DEFAULT group default 
    link/ether 6e:b4:f0:57:18:e4 brd ff:ff:ff:ff:ff:ff link-netnsid 3
8: vethcd3a89c@if7: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP mode DEFAULT group default 
    link/ether 66:ca:cf:ce:10:a9 brd ff:ff:ff:ff:ff:ff link-netnsid 4
1846: eth0@if1847: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue state UP mode DEFAULT group default 
    link/ether 02:42:ac:14:14:05 brd ff:ff:ff:ff:ff:ff link-netnsid 0
```

What am I missing here?

Reply:
ah... I think I got it
the datapath containers have certain names, hence they hold the clab interfaces, not the parent container.

Reply:
@hellt indeed. As soon as eth1/eth2 is created inside ixia-c-one (parent docker:dind container), a new traffic-engine container is spawned and the corresponding interface is pushed to its network namespace.

Hence those interfaces can be found inside, `docker exec -it <ixia-c-one container> bash` -> `docker ps -a` -> `docker exec -it <traffic engine container> bash`